Backend/app.py

```diff
@@ -38,6 +121,38 @@ def chat():
         if not message:
             return jsonify({"error": "Message cannot be empty"}), 400
 
+        # Get feature-flagged configuration for this request
+        targeting_context = targeting_accessor(request)
+        
+        # Get AI endpoint variant
+        ai_endpoint = None
+        agent_id = None
+        
+        if feature_manager:
+            ai_endpoint_variant = feature_manager.get_variant(
+                feature_flag="ai_endpoint",
+                targeting_context=targeting_context
+            )
+            ai_endpoint = ai_endpoint_variant.configuration if ai_endpoint_variant else config.get("ai_endpoint")
+            
+            # Get Agent variant
+            agent_variant = feature_manager.get_variant(
+                feature_flag="Agent", 
+                targeting_context=targeting_context
+            )
```

Comment:
get_variant takes in a string that is the variant name and feature_manager should already have the targeting context from the init.